File: programming/3.2/main.cpp
#include <iostream>
#include <cmath>


using namespace std;

float f1(int x) {
  return 3 + x / (x + 3);
}

float f2(int x) {
  return pow(2 - x, 3) + pow(x, -4) + 3 * pow(x, 1/2);
}

float f3(int x) {
  return pow(sin(x), 2) + 25 * cos(x) - 2 / (pow(x, -3) + 2);
}

int main() {
  float a, z, b = 1, x = 10;

  for (int i = 0; i < 16 + 8; i++) {
    a += f2(x);
  }

  for (int i = 0; i < 16 + 8; i++) {
    b = b * f3(x);
  }

  cout << "Result a: " << a << endl;
  cout << "Result b: " << b << endl;

  z = exp(3 * a * b);

  cout << "Reslut z: " << z << endl;
  
  return 0;
}

File: programming/3.2/main.py
from math import sin, cos, exp
from prettytable import PrettyTable


# v 16
table = PrettyTable()
table.field_names = ['a', 'b', 'z']

def f1(x: int) -> float:
  return 3 + x / (x + 3)

def f2(x: int) -> float:
  return (2 - x)**3 + x**-4 + 3 * x**1/2

def f3(x: int) -> float:
  return sin(x)**2 + 45 * cos(x) - 2 / (x**3 + 2)

a = 0
b = 1
z = 0
x = 1

for i in range(16 + 8 + 1):
  a += f2(x)
  b *= f3(x)
  z = 3 * a * b
  table.add_row([a, b, '%.2f' % z])

print(table)